File: advanced/old_xml-and-json/src/main/java/org/tsdes/spring/xmlandjson/data/TopPosts.java
package org.tsdes.spring.xmlandjson.data;

import javax.xml.bind.annotation.*;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
public class TopPosts {

    @XmlElement(required = true)
    private Date retrievedTime;

    @XmlElementWrapper(name = "postList", required = true)
    @XmlElement(name = "post", required = true)
    private List<Post> posts;


    public void addPost(Post p){
        if(posts == null){
            posts = new ArrayList<>();
        }
        posts.add(p);
    }


    public Date getRetrievedTime() {
        return retrievedTime;
    }

    public void setRetrievedTime(Date retrievedTime) {
        this.retrievedTime = retrievedTime;
    }

    public List<Post> getPosts() {
        return posts;
    }

    public void setPosts(List<Post> posts) {
        this.posts = posts;
    }






}

File: advanced/data-format/src/main/kotlin/org/tsdes/advanced/dataformat/data/TopPosts.kt
package org.tsdes.advanced.dataformat.data

import java.util.*
import javax.xml.bind.annotation.*;

@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
data class TopPosts(

        @field:XmlElement(required = true)
        var retrievedTime: Date? = null,

        @field:XmlElementWrapper(name = "postList", required = true)
        @field:XmlElement(name = "post", required = true)
        var posts: MutableList<Post>? = null
) {

    fun addPost(p: Post) {
        if (posts == null) {
            posts = ArrayList()
        }
        posts!!.add(p)
    }
}